test/test_xla_sharding.py

```diff
@@ -11,38 +11,60 @@
 import torch_xla.utils.utils as xu
 import torch_xla.experimental.xla_sharding as xs
 from torch_xla.experimental.xla_sharded_tensor import XLAShardedTensor
+from torch_xla.experimental.pjrt import using_pjrt
 
 
-@unittest.skipIf((os.getenv('PJRT_DEVICE') == "") or (
-    xm.get_xla_supported_devices("GPU") is not None
-), "PyTorch/XLA SPMD requires PJRT_DEVICE={CPU, TPU}, GPU is currently not supported."
-                )
+@unittest.skipIf(not using_pjrt() or xm.get_xla_supported_devices("GPU"),
+                 f"Requires PJRT_DEVICE set to `TPU` or `CPU`.")
 class XlaShardingTest(unittest.TestCase):
 
+  n_devices = 0
+  device_ids = None
+
+  @classmethod
+  def setUpClass(cls):
+    cls.n_devices = len(xm.get_xla_supported_devices())
+    cls.device_ids = np.array(range(cls.n_devices))
+
+  def _get_mesh(self, mesh_shape, device_ids=None):
+    if device_ids is None:
+      device_ids = self.device_ids
+    assert len(device_ids) == self.n_devices
+    return xs.Mesh(device_ids, mesh_shape)
+
   def test_xla_sharded_tensor(self):
-    n_devices = xm.xrt_world_size()
-    mesh_shape = (1, n_devices)
     partition_spec = (0, 1)
     xt1 = torch.tensor([[1, 2, 3, 4, 5, 6, 7, 8]],
                        dtype=torch.float,
                        device=xm.xla_device())
-    xst1 = xs.mark_sharding(xt1, mesh_shape, partition_spec)
+    xst1 = xs.mark_sharding(xt1, self._get_mesh((1, self.n_devices)),
+                            partition_spec)
 
     # TODO(244003536) add more tests for XLAShardedTensror.
     self.assertTrue(isinstance(xst1, XLAShardedTensor))
 
+  def test_custom_tile_assignment(self):
+    xt = torch.randn(10, 20).to(device=xm.xla_device())
+    mesh_shape = (1, self.n_devices)
```

Comment:
I see the tests have all devices mapped to a single axis - is there anything stopping us from using e.g. `mesh_shape = (2, self.n_devices / 2)`?

Reply:
No, but for the unit testing a flat mesh is easier to work with since we don't know how many devices we would have (e.g., for CPU, we will have 1).